.github/workflows/pull-request-ci.yaml

```diff
@@ -0,0 +1,307 @@
+name: CI
+on: push
```

Comment:
I don't think this does what the description asks for.
Please see [1]. We need this to run for checking PRs, but not those that are opened from forks. Just from local branches.

[1] https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#on

Reply:
Yep, I did it on push in my branch for spiking; this will need to change before we actually merge.

Reply:
done